### PR TITLE
Changes for KNIDEPLOY-2111

### PR DESCRIPTION
**Pull request by aireilly**
# Description

If the MAC address of an existing ironic node and a Bare Metal host matches, then the operator will associate the two. When the host enrollment, inspection, cleaning, or other ironic steps fail, the baremetal-operator will continuously retry. Need to let users know it will retry.

Documents #KNI-2111

## Type of change

Please select the appropriate options:

- [x] This change is a documentation update

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [ ] A change should not be merged unless it passes CI or there is a comment/update saying what testing was passed.
- [ ] PRs should not be merged unless positively reviewed.

### CODE REVIEW

**Review by iranzo — APPROVED** [review]
/lgtm
/approve

**Review by johnwilkins — APPROVED** [review]
LGTM. However, I sort of loath using both the term "node" and the term "host" in the same sentence. I wish there were a better way of stating this. Linux expressly uses the term host and hostname for example. I prefer node to host and server, but it's a personal thing.

**Comment by iranzo** [comment]
> LGTM. However, I sort of loath using both the term "node" and the term "host" in the same sentence. I wish there were a better way of stating this. Linux expressly uses the term host and hostname for example. I prefer node to host and server, but it's a personal thing.

do a /hold cancel  once it's ready for merging

**Comment by aireilly** [comment]
@johnwilkins @iranzo John to your point - in this instance, and in OCP, a "bare metal node" is explicilty a host which has been provisioned as a machine in the cluster, while a "bare metal host" has not yet been provisioned as a "bare metal node" in the cluster. Correct me if I am wrong /honk

**Comment by aireilly** [comment]
@johnwilkins @iranzo I've added some more detail on how to diagnose a dupe MAC address when adding a host to the cluster, it's in a new module.

**Review by iranzo — CHANGES_REQUESTED** [review]
Hi, if the PR is no longer WIP, edit the title to remove it, if not, it will not be mergeable.

Also, @johnwilkins WDT? anything else or can be merged?

Thx

**Comment by aireilly** [comment]
Waiting for final review from @johnwilkins before merge.

File: documentation/ipi-install/modules/ipi-install-diagnosing-duplicate-mac-address.adoc
Comment (johnwilkins) [inline documentation/ipi-install/modules/ipi-install-diagnosing-duplicate-mac-address.adoc:5]: s/operator/Operator/  The Operator displays a registration error.

File: documentation/ipi-install/modules/ipi-install-diagnosing-duplicate-mac-address.adoc
Comment (johnwilkins) [inline documentation/ipi-install/modules/ipi-install-diagnosing-duplicate-mac-address.adoc:17]: `openshift-machine-api`

**Review by johnwilkins — COMMENTED** [review]
Just a few comments.

**Comment by aireilly** [comment]
> Hi, if the PR is no longer WIP, edit the title to remove it, if not, it will not be mergeable.
> 
> Also, @johnwilkins WDT? anything else or can be merged?
> 
> Thx

This is ready to merge now.

**Pull request merged**
